### DISCUSSION TITLE
i2c_speed global or for external interfaces ?

### DISCUSSION THREAD

**Discussion by mrechte** [iNavFlight/inav#7979] [Q&A] [answered]
Hi.
I am not clear to which interfaces relates this setting. For instance a speedybee F7 V2  stages internal MPU8000 (gyro+accel) and BMP280 (baro) with i2c speed, respectively 400KHz and 3.4MHz. On top of that, I have an external HMC5883L (magneto) at 400KHz.
Should I set the speed to the min, viz. 400KHz ?
Thanks

**Comment by DzikuVx** [comment, marked as answer]
I2C Speed affect all I2C buses in use. MPU6000 (not 8000) is SPI, so it does not apply

**Reply by mrechte** [reply]
Thanks a lot.
